src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQueryAvroToStructuredTransformer.java

```diff
@@ -103,4 +104,14 @@ protected Object convertField(Object field, Schema fieldSchema) throws IOExcepti
     }
     return super.convertField(field, fieldSchema);
   }
+
+  @Override
+  protected Object convertBytes(Object field) {
+    if (field instanceof ByteBuffer) {
+      ByteBuffer buf = (ByteBuffer) field;
+      return Arrays.copyOfRange(buf.array(), buf.arrayOffset() + buf.position(),
```

Comment:
Use `Bytes.toBytes(buf)` from cdap-api instead. It works for all types of ByteBuffer implementations.